packages/connection-storage/src/connection-storage.spec.ts

```diff
@@ -75,6 +87,238 @@ describe('ConnectionStorage', function () {
     }
   });
 
+  describe('migrateToSafeStorage', function () {
+    let sandbox: Sinon.SinonSandbox;
+    beforeEach(function () {
+      sandbox = Sinon.createSandbox();
+    });
+
+    afterEach(function () {
+      sandbox.restore();
+    });
+
+    context('does not migrate connections', function () {
+      it('when there are no connections', async function () {
+        await ConnectionStorage.migrateToSafeStorage();
+        const connections = await ConnectionStorage.loadAll();
+        expect(connections).to.deep.equal([]);
+      });
+
+      it('when there are only legacy connections', async function () {
+        await writeFakeConnection(tmpDir, connection1270);
+
+        const encryptSecretsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'encryptSecrets'
+        );
+        const getKeytarCredentialsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'getKeytarCredentials'
+        );
+
+        await ConnectionStorage.migrateToSafeStorage();
+
+        expect(
+          encryptSecretsSpy.called,
+          'it does not try to encrypt'
+        ).to.be.false;
+        expect(
+          getKeytarCredentialsSpy.called,
+          'it does not try to get secrets from keychain'
+        ).to.be.false;
+      });
+
+      it('when there are only migrated connections', async function () {
+        const connectionInfo1 = getConnectionInfo();
+        const connectionInfo2 = getConnectionInfo();
+
+        await writeFakeConnection(tmpDir, {
+          _id: connectionInfo1.id,
+          connectionInfo: connectionInfo1,
+          version: 1,
+        });
+        await writeFakeConnection(tmpDir, {
+          _id: connectionInfo2.id,
+          connectionInfo: connectionInfo2,
+          version: 1,
+        });
+
+        const encryptSecretsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'encryptSecrets'
+        );
+        const getKeytarCredentialsSpy = sandbox.spy(
+          ConnectionStorage,
+          <any>'getKeytarCredentials'
```

Comment:
```suggestion
          'getKeytarCredentials' as any
```

we generally don't do the `<...>` syntax because that's a) inconsistent with the rest of the codebase and b) has broken some scripts in the past (depcheck at least, iirc)

Reply:
fixed in [df48b1b](https://github.com/mongodb-js/compass/pull/5115/commits/df48b1bffc7ac8136fe35a45425aca2e190d7a9c)